gcs-bigquery-loader/app/main.py

```diff
@@ -0,0 +1,61 @@
+from google.api_core.exceptions import Conflict
+from google.cloud import bigquery
+import os
+
+
+def load_avro_to_bigquery(data, context):
+    """Background Cloud Function triggered by Cloud Storage `finalize` events.
+        Loads newly created Avro files to BigQuery.
+
+        This function expects following directory structure on GCS:
+        `{namespace}/{doc_type}/{version}/{file_name}.avro`
+        
+        Each file is appended to a corresponding table, its name constructed in the following way:
+        `{namespace}_{doc_type}_{version}`
+        Table needs to exist, otherwise the load job will fail and will not be retried. In such
+        cases manual backfill will be required.
+
+    Args:
+        data (dict): The Cloud Functions event payload.
+        context (google.cloud.functions.Context): Metadata of triggering event.
+    Returns:
+        None; asynchronous BigQuery load job is started, 
+        the output is written to Stackdriver Logging
+    """
+
+    dataset = os.environ.get("BIGQUERY_DATASET", "default_dataset")
+
+    bucket = data["bucket"]
+    file_path = data["name"]
```

Comment:
This looks like we're getting one invocation of the function per file uploaded to GCS. Is that correct?

I'd expect Beam's FileIO to output tens or hundreds of files (whatever numShards is set to) per table per 5 minutes, so I'd expect a load-per-file strategy to quickly run into the per-table daily load job limit.

Reply:
That's correct, it's called every time a file is finalized. If the job is going to dump hundreds of files per batch, that's not a good solution indeed.